src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,15 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(!fNRows && !fNCols && fNCols!=fNRows )
+        return;
+    for (size_t i = 0; i < fNCols; ++i) {
+        fArmaMat(i,i)=1;
```

Comment:
don't you need to worry about the off diagonal elements? If the matrix isn't zero to begin with you won't get the identity